pkg/pending/cache.go

```diff
@@ -0,0 +1,105 @@
+package pending
+
+import (
+	"context"
+	"sort"
+	"sync"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log = logf.Log.WithName("pending_object_cache")
+
+type GetListOfPendingObjects func(cl client.Client, labelListOption client.ListOption) ([]client.Object, error)
+
+type cache struct {
+	sync.RWMutex
+	objectsByCreation       []string
+	client                  client.Client
+	objectType              client.Object
+	getListOfPendingObjects GetListOfPendingObjects
```

Comment:
```suggestion
	listPendingObjects ListPendingObjects
```
(due to renaming above)

Reply:
renamed af36438